isthmus/src/main/java/io/substrait/isthmus/expression/ExpressionRexConverter.java

```diff
@@ -178,6 +186,20 @@ public RexNode visit(Expression.IntervalYearLiteral expr) throws RuntimeExceptio
         new BigDecimal(expr.years() * 12 + expr.months()), YEAR_MONTH_INTERVAL);
   }
 
+  private static final long MICROS_IN_DAY = TimeUnit.DAYS.toMicros(1);
+
+  @Override
+  public RexNode visit(Expression.IntervalDayLiteral expr) throws RuntimeException {
+    return rexBuilder.makeIntervalLiteral(
+        // Current Calcite behavior is to store milliseconds since Epoch
+        // microseconds version: new BigDecimal(expr.days() * MICROS_IN_DAY + expr.seconds() *
+        // 100000L + expr.microseconds()), DAY_SECOND_INTERVAL);
```

Comment:
Is that Calcite behaviour part of the type system? What would be involved in allow microsecond precision as per the spec (if you happen to know).

Reply:
Both Avatica and Calcite use Epoch milliseconds from timestamps and milliseconds for day-time intervals. This is a huge refactoring which will probably imply a major version change if it happens.

Reply:
We can do that as a future change then.